pkg/controller/flight/reconciler.go

```diff
@@ -39,18 +42,70 @@ func (f *flightReconciler) AddMissingSecurityGroup() []string {
 			continue
 		}
 
-		if err := f.Client.SetSecurityGroup(node.GetID()); err != nil {
+		if err := f.Client.SetSecurityGroup(instance.GetID()); err != nil {
 			f.Logger.Log(
 				"msg", "couldn't set securitygroup",
-				"node_id", node.GetID(),
+				"instance", instance.GetID(),
 				"err", err)
 			continue
 		}
-		ids = append(ids, node.GetID())
+		ids = append(ids, instance.GetID())
 	}
 	return ids
 }
 
 func (f *flightReconciler) DeleteIncompletelySpawnedInstances() []string {
-	return []string{}
+	deletedInstanceIDs := []string{}
+	timedOutInstances := f.getTimedOutInstances()
+	unregisteredInstances := f.getUnregisterdInstances()
+
+	for _, unregistered := range unregisteredInstances {
+		found := false
+		for _, timedOut := range timedOutInstances {
+			if unregistered.GetName() == timedOut.GetName() {
+				found = true
+				break
+			}
+		}
+
+		if found {
+			if err := f.Client.DeleteNode(unregistered.GetID()); err != nil {
```

Comment:
This could in deed happen when an instance is stuck in deleting. I'm unsure we should be handling that. It sounds more like a job for a Nova nanny.

If an error occurs this will try to delete it again in the next reconciliation loop. Also, I tried to write funcs to be unaffected by errors and not break the global loop. Hence this only logs the error and `continue`s. It should at most result in 1 log line every 5 minutes, if an instance is stuck or a concurrent deletion happens.